### PR TITLE
Feat-CNN

### PR DESCRIPTION
**Pull request by AtharvaPatange**
Overview
This implementation defines a Convolutional Neural Network (CNN) in R using the Keras library. The CNN is designed to automatically extract spatial and hierarchical features from image data for tasks such as image classification, object detection, and visual recognition. The model consists of convolutional, pooling, flatten, and dense layers, with ReLU and Softmax activations.

Features

Automatically extracts local spatial patterns from images using convolutional layers.
Reduces spatial dimensions and retains important features with max-pooling layers.
Adds non-linearity using ReLU activations to capture complex patterns.
Converts 2D feature maps into 1D vectors for fully connected processing.
Dense layers combine extracted features for final classification.
Output layer provides class probabilities using Softmax activation.
Compatible with image datasets such as MNIST and CIFAR-10.
Fully modular architecture that can be expanded with additional layers.

Complexity

Time Complexity: O(E × N × F × K²), where E = epochs, N = number of samples, F = number of filters, K = kernel size

Space Complexity: O(parameters + feature maps)

Demonstration

The included R script defines the CNN model and displays its summary.
To train the model, use model %>% compile() followed by model %>% fit() on an image dataset.
Example datasets: MNIST (28×28 grayscale images) or CIFAR-10 (32×32 RGB images).

Summary
This implementation provides a fully documented CNN architecture in R for image-based tasks. It is suitable for automatic feature extraction and classification and serves as a foundation for building more complex deep learning models in computer vision.

### CODE REVIEW

File: dynamic_programming/kadane's_algo.r [hunk 1]
```diff
@@ -0,0 +1,162 @@
+# Kadane's Algorithm in R
```
[the remark this answers is not in the document]
Comment (siriak) [reply dynamic_programming/kadane's_algo.r:1]: Please remove the '

**Review by siriak — CHANGES_REQUESTED** [review]
